cli/src/semgrep/app/scans.py

```diff
@@ -419,11 +425,18 @@ def report_findings(
         except requests.RequestException as exc:
             raise Exception(f"API server returned this error: {response.text}") from exc
 
-        try_until = datetime.now() + timedelta(minutes=20)
-        slow_down_after = datetime.now() + timedelta(minutes=2)
         complete_task = progress_bar.add_task("Finalizing scan")
-        while datetime.now() < try_until:
-            logger.debug("Sending /complete")
+        try_until = datetime.utcnow() + timedelta(minutes=20)
+        slow_down_after = datetime.utcnow() + timedelta(minutes=2)
```

Comment:
nit: It feels unnecessary to have this based on time. Could we do something like:
```
num_quick_polls = [5] * 24  # 24 attempts with 5s backoff, 120s total
num_long_polls = [30] * 36 # 36 attempts with 30s backoff, remaining 18 mins
poll_attempts = num_quick_polls + num_long_polls
for sleep_secs in poll_attempts:
    # attempt to complete
    # if success, return result
    sleep(sleep_secs)
```

Reply:
I did it based on time because it makes the code more predictable, requests can take up to 60s, get internally retried up to 3 times, so a single loop might consume 3 minutes in the worst case.

Reply:
Ahhh ok that makes sense.